src/ChannelServer/World/Inventory.cs

```diff
@@ -209,19 +220,62 @@ public Item GetItem(EquipSlot slot)
 		/// </summary>
 		/// <param name="item"></param>
 		/// <return>Index of the item.</return>
-		public int AddSilent(Item item)
+		public Dictionary<int, int> AddSilent(Item item)
 		{
 			var cat = item.Data.Category;
-
 			lock (_syncLock)
 			{
+				Dictionary<int, int> itemsUpdated = new Dictionary<int, int>();
```

Comment:
Just a tip, if you use the [var keyword](https://msdn.microsoft.com/en-us/library/bb383973.aspx), you don't need to type as much, the compiler tries to figure out the type itself then, based on what you put in. E.g. `var itemsUpdated = new Dictionary<int, int>();`